src/main.ts

```diff
@@ -100,6 +110,177 @@ export class Minifier {
     return path.join(destination, subFilePath);
   }
 
+  isExternal(symbol: ts.Symbol): boolean {
+    // TODO: figure out how to deal with undefined symbols
+    // (ie: in case of string literal, or something like true.toString())
+    if (!symbol) return true;
+
+    return symbol.declarations.some((decl) => !!(decl.getSourceFile().fileName.match(/\.d\.ts/)));
+  }
+
+  isRenameable(symbol: ts.Symbol): boolean {
+    if (this.isExternal(symbol)) return false;
+    if (!this._typeCasting.has(symbol)) return true;
+
+    let boolArrTypeCasting = [];
+    let renameable = true;
+
+    // Three cases to consider:
+    // CANNOT RENAME: Use site passes an internally typed object, expected site wants an externally
```

Comment:
or vise versa that is 'use site passes externally typed obj, but expected site wants an internally typed object'